### PR TITLE
feature: init CLI command

### LINKED ISSUE #222: [FEATURE] Reassure init CLI command
**Is your feature request related to a problem? Please describe.**
Add `init` command to CLI. This comment would be invoked like this:
```ssh
$ yarn reassure init
```

This command would apply common Reassure setup steps that need to be done by hand now:
- create `reassure-tests.sh` script from [template](https://github.com/callstack/reassure#write-performance-testing-script)
- create dangerfile from [template](https://github.com/callstack/reassure/blob/main/dangerfile.ts)
- add `.reassure` folder to `.gitignore` if present
- show link how to add reassure to CI: https://github.com/callstack/reassure#ci-integration

**Describe the solution you'd like**
Before doing the changes mentioned above tool should check if `reassure-tests.sh` exists, and assume that if it exists then the folder hand been already initialized for Reassure

For subsequent steps (danger file, gitignore): they should be only carried if danger file does not yet exist, or if gitignore does not contain reassure entry.

The tool should be user friendly and have clear help messages.

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Resolves #222 

Based on code extracted from #275 

image

Scope:
- [x] `reassure init` CLI command
- [x] docs updates

Tasks:
- [x] Dangerfile TS/JS detection

### Test plan

All checks should pass.

### CODE REVIEW

**Review by Xiltyn — APPROVED** [review]
I think everything is extracted as was essentially. So, looks good to me :) Thanks for doing the ground work of extracting it from the previous PR.

**Comment by Xiltyn** [comment]
I have a question though. Why do you think all of the tests dropped in performance? Just a flake?

wdyt? @mdjastrzebski

**Comment by mdjastrzebski** [comment]
@Xiltyn I want to polish some rough edges, I will make let you know when PR is ready for review.

re perf. changes: this should be purely random fluctuations.

**Comment by mdjastrzebski** [comment]
@Xiltyn I've finished tweaking the PR. List of changes:
- [x] Streamline the init process by checking the relevant steps 
- [x] Update the `init` output so that it is similar to existing output
- [x] Streamline Dangerfile creation, detection of existing Dangerfile.js/ts,  removal of `--javascript` option

File: packages/reassure-cli/src/commands/init.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  DANGERFILE_FALLBACK_TS,
+  DANGERFILE_JS,
+  DANGERFILE_TS,
+  GIT_IGNORE,
+} from '../constants';
+import { applyCommonOptions, CommonOptions } from '../options';
+import { ASCII_BYE, ASCII_HELLO } from '../utils/ascii';
+import { configureLoggerOptions } from '../utils/logger';
+
+const TEMPLATE_PATH = `${__dirname}/../templates`;
```
Comment (thymikee) [inline packages/reassure-cli/src/commands/init.ts:17]: Can you use path joining for windows compatibility?

File: packages/reassure-cli/src/commands/init.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    logger.log(`🔗 ${resolve(existingFile)}`);
+    return;
+  }
+
+  logger.clearLine();
+  logger.log(`⚠️   Dangerfile: created ${fallbackFile} - merge with existing ${existingFile}`);
+  logger.log(`🔗 ${resolve(fallbackFile)}`);
+}
+
+function queryDangerfile(): [string, string] | [null, null] {
```
Comment (thymikee) [inline packages/reassure-cli/src/commands/init.ts:93]: Shouldn't we prompt the user whether they want to use Danger or not? Itcould include description of what it is

Comment (mdjastrzebski) [reply packages/reassure-cli/src/commands/init.ts:93]: Atm we do not support non-danger integrations, so Danger Setup is Linda necessary. When we add any alternative, we can prompt the user about it.

Comment (thymikee) [reply packages/reassure-cli/src/commands/init.ts:93]: You can use Reassure without comment reporter. It's only a recommended setup, but not strictly required to run the comparison on a CI

**Review by Xiltyn — APPROVED** [review]
In general I think that's very much the baseline of the feature and everything looks great.

However personally, I'm not sure why are we removing prompts from the init script. I understand removing the interactive options for starting other scripts as this is complicating things but from init as well? 

For example, I'm not sure that defaulting to JS, instead of asking the user is a good user experience necessarily, rather feels a little bit sloppy on the UX side of things. And there has already been a point raised about danger too. 

So, while I don't think it's crucial for this PR and I don't want to block it, I do think it is ill advised for us not to deliver the best experience we can. Therefore, I would be against disregarding this idea entirely.

**Comment by mdjastrzebski** [comment]
Let's add prompts separately. Currently the damage to the user isn't big, they can simply delete dangerfile if not needed.

**Pull request merged**